### PR TITLE
Docs: how to use kwargs with custom task graphs

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
This PR adds information to the docs about how to use custom task graphs with functions that use keyword arguments. 

Somebody pleas check my RST formatting, a second set of eyes would be great.

This is a documentation update, so I haven't written a test. (Honestly it would be amazing if we could somehow make all the custom task graph examples here runnable in ipython, but that is beyond the scope of this PR)

- [x] Closes https://github.com/dask/dask/issues/9321
- [ ] ~~Tests added / passed~~
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by douglasdavis** [comment]
Perhaps `dask.utils.apply` should be added to the list of utilities at https://github.com/dask/dask/blob/main/docs/source/api.rst#utilities; then referenced in the new section

File: docs/source/custom-graphs.rst [hunk 1]
```diff
@@ -56,6 +56,30 @@ analyze pipeline:
    get(dsk, 'store')  # executes in parallel
 
 
+
+Keyword arguments in custom Dask graphs
+---------------------------------------
+
+Sometimes, you may want to pass keyword arguments to a function 
+in a custom Dask graph.
```
Comment (douglasdavis) [inline docs/source/custom-graphs.rst:64]: Maybe add another sentence mentioning `dask.utils.apply`, and (give it a
```
:func:`dask.utils.apply`
```
link) if that can get added to the list at https://github.com/dask/dask/blob/main/docs/source/api.rst#utilities

Comment (GenevieveBuckley) [reply docs/source/custom-graphs.rst:64]: Nice suggestion, done!

File: docs/source/custom-graphs.rst [hunk 2]
```diff
@@ -56,6 +56,31 @@ analyze pipeline:
...
+You can do that using the :func:`dask.utils.apply`, like this:
```
Comment (douglasdavis) [inline docs/source/custom-graphs.rst:65]:
```suggestion
You can do that using the :func:`dask.utils.apply` function, like this:
```

**Review by douglasdavis — COMMENTED** [review]
Now in the docs preview `dask.utils.apply` looks a bit underwhelming 🙃 :
Screen Shot 2022-07-28 at 9 18 39 AM
What you have here for the .rst page:
```
In the example above:

- ``args`` should be a tuple (eg: ``(arg_1, arg_2, arg_3)``), and
- ``kwargs`` should be a dictionary (eg: ``{"kwarg_1": value, "kwarg_2": value}``
```
Looks like a good start to a docstring

**Comment by ncclementi** [comment]
@GenevieveBuckley what do you think of adding @douglasdavis suggestion to the docstring, so the api docs don't look empty? See link https://dask--9322.org.readthedocs.build/en/9322/api.html#utilities

**Comment by GenevieveBuckley** [comment]
I think it's a good idea. The delay on my end is because:
1. I was trying to construct a good simple docstring example, and
2. I'm not sure how to say you shouldn't need to ever use this function as a beginner to intermediate user of Dask and that it's mostly good for injecting keyword arguments into low level task graphs.

**Comment by GenevieveBuckley** [comment]
Ok, I've tried to explain in the docstring that most people will never need to use this function. Hopefully this is clear enough!

File: dask/utils.py [hunk 3]
```diff
@@ -37,6 +37,36 @@
...
+    Examples
+    --------
+    >>> from dask.utils import apply
+    >>> def add(number, second_number=5):
+    ...     return number + second_number
+    ...
+    >>> apply(add, (10,), {"second_number": 2})  # equivalent to func(*args, **kwargs)
+
+    >>> from dask.utils import apply
+    >>> task = (apply, func, args, kwargs)  # equivalent to func(*args, **kwargs)
```
Comment (jsignell) [inline dask/utils.py:67]: I changed the example around a little to make it pass doctests, but I think this gets at the same concept:
```suggestion
    >>> apply(add, (10,), {"second_number": 2})  # equivalent to add(*args, **kwargs)
    12

    >>> task = apply(add, (10,), {"second_number": 2})
```

Comment (GenevieveBuckley) [reply dask/utils.py:67]: Thank you!

**Comment by jsignell** [comment]
Thanks @GenevieveBuckley! Love this docs work :)

**Pull request merged**